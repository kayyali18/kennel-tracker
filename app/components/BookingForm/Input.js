import React from 'react'
import PropTypes from 'prop-types'
import InputWrapper from './InputWrapper'

function Input(props) {
  return (
    <InputWrapper>
      <input
        onChange={props.handleChange}
        id={props.id}
        autoComplete="true"
        required
        type={props.type}
        placeholder={props.placeholder}
      />
    </InputWrapper>
  )
}

export default Input

Input.propTypes = {
  name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.string,
  placeholder: PropTypes.string,
}
