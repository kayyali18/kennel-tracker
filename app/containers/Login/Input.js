import styled from 'styled-components'

const Input = styled.input`
  display: block;
  outline: none;
  margin: 80px auto;
  padding: 8px;
  background: #fff;
  border: 1px solid grey;
`
export default Input
