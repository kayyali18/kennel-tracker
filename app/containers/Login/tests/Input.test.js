import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Input from '../Input'

describe('<Input />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderer.create(<Input />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should render an input', () => {
    const renderedComponent = renderer.create(<Input />).toJSON()
    expect(renderedComponent.type).toEqual('input')
  })
})
