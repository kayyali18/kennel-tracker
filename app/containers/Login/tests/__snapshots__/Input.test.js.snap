// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
.c0 {
  display: block;
  outline: none;
  padding: 25px;
  size: 2em;
  color: black;
  margin: 10px;
  border: 1px solid black;
}

<input
  className="c0"
/>
`;
